### DISCUSSION TITLE
Projecting Fluidity fields on different order meshes

### DISCUSSION THREAD

**Discussion by gnikit** [FluidityProject/fluidity#339] [Q&A]
I was wondering if there is an easy way to project, using Fortran and not the schema, a Fluidity field (for the sake of this discussion let's make the field scalar) from a P1 mesh onto a P2 mesh and vice versa.

I know we have the `Superconvergence` and `Supermesh_Assembly` femtools modules which I suspect should be able to do the trick but they are not very well documented, or I am doing something silly, so I am not entirely sure how to achieve the projection.

**Comment by Patol75** [comment]
I am terribly late to the party, but if you are looking at a scalar or vector field defined within the FLML, then you can set an equivalent scalar or vector field as diagnostic and use the scalar/vector_galerkin_projection algorithms. However, I am not too sure how to proceed if you need to deal with a tensor field.

I will admit I did not fully understand what you meant by "using Fortran and not the schema", so apologies if the above does not specifically target your query.

**Reply by gnikit** [reply]
No worries @Patol75 I actually don't recall if I ever resolved this issue. The fields in question are always scalar but are not part of the schema, they are generated internally and at some point in the execution loop (after a solve and before adaptivity) you want to project that field from its mesh to a higher order mesh, do some basic computations there, and then project field back down to its original mesh.

I was looking for a way of doing that programmatically in Fortran (hence my comment) with, if possible, the routines that Fluidity already defines. I vaguely recall using a very bad projection algorithm I conjured up during an evening and surprise-surprise the results were terrible.
